File: module/php/posty.php
<?php

/**
 * @class Posty
 * @author Adam Timberlake
 */
class Posty {

    /**
     * @constant API_URL
     * @type string
     */
    const API_URL = 'http://api.zoopla.co.uk/api/v1/property_listings.js?postcode=%s&api_key=%s';

    /**
     * @constant API_KEY
     * @type string
     */
    const API_KEY = 'nrb28jhb8vydfbf7769fkat3';

    /**
     * @method getLatLng
     * @param string $postCode
     * @return array
     */
    public function getLatLng($postCode) {

        // Fetch the data and then decode the JSON.
        $url  = sprintf(self::API_URL, $postCode, self::API_KEY);
        $data = file_get_contents($url);
        $data = (object) json_decode($data);

        // Calculate the latitude/longitude from the bounding box.
        $boundingBox   = $data->bounding_box;
        $longitudeDiff = $boundingBox->longitude_max - $boundingBox->longitude_min;
        $longitude     = $boundingBox->longitude_max - $longitudeDiff;
        $latitudeDiff  = $boundingBox->latitude_max - $boundingBox->latitude_min;
        $latitude      = $boundingBox->latitude_max - $latitudeDiff;

        return array('latitude' => $latitude, 'longitude' => $longitude);

    }

}

$posty    = new Posty();
$postCode = urlencode($_GET['postCode']);
echo json_encode($posty->getLatLng($postCode));

File: module/ruby/Posty.rb
require 'json'
require 'open-uri'

#
# @class Posty
# @author Adam Timberlake
#
class Posty

  #
  # @constant ZOOPLA_API_URL
  # @type string
  #
  ZOOPLA_API_URL = 'http://api.zoopla.co.uk/api/v1/property_listings.js?postcode=%s&api_key=%s'

  #
  # @constant ZOOPLA_API_KEY
  # @type string
  #
  ZOOPLA_API_KEY = 'nrb28jhb8vydfbf7769fkat3'

  #
  # @constant NOMINATIM_API_URL
  # @type string
  #
  NOMINATIM_API_URL = 'http://nominatim.openstreetmap.org/search?format=json&limit=5&q=%s&addressdetails=1'

  #
  # @property cache
  # @type string
  # @default nil
  #
  @cache = nil

  #
  # @property cache_document
  # @type string
  # @default nil
  #
  @cache_document = nil

  #
  # @method get_lat_lng
  # @param post_code string
  # @return string
  #
  def get_lat_lng(post_code)

  end

  #
  # @method set_cache
  # @param document string
  # @return nil
  #
  def set_cache(document)
    data = File.read document
    @cache = !data.empty? ? JSON.parse(data) : []
    @cache_document = document
  end

end

posty = Posty.new
posty.set_cache('../../cache/posty.json')
postCode = URI::encode 'NG10 4FP'
puts posty.get_lat_lng(postCode).to_json